### DISCUSSION TITLE
couchdb_httpd_up metric suggestion

### DISCUSSION THREAD

**Discussion by equinsuocha** [gesellix/couchdb-prometheus-exporter#92] [General]
Hi there, I have some trouble understanding `couchdb_httpd_up` metric, so description tells us that `Was the last query of CouchDB stats successful.`, but if we look at exporter's code it's quite a complicated process with many possible points of failure, and practically when node stats collection fails for any reason (for example, in my case this error tends to happen from time to time: https://github.com/gesellix/couchdb-prometheus-exporter/issues/30), I can see a strange situation when `couchdb_httpd_up == 0`, but `couchdb_httpd_node_up == 1` for every `node_name`, which makes wrong impression that service is down, however, actually it reflects some internal processing error which does not even affect node statuses and interrupt any other metric collection, and this metric is only showing that there was some error along the way and does not seem that important;
So my question is, mb it makes sense to either change it to some general api health check, i.e. get "http://couch:5984" and just check response code, so we know that API is reachable and then if necessary introduce some internal metrics helping to understand that smth went wrong during `stats` collection, for example internal error counter labeled with error codes.

If some of that makes sense and we can agree conceptually, I would like to contribute, since we are actively using your exporter in our enviromnent.

**Comment by gesellix** [comment]
Hey, thanks for pointing on that issue I was unaware of.

I haven't had thorough thoughts about that behavior, so this is only a rough direction I'd like to go: the _up metric should express availability and also help understanding if the metrics in the response are complete and to be trusted.

Yet, I do understand that "up" usually is more like a health check and should be simple and reliable.

If you'd like to contribute a change, I'd prefer something like that:

- simplify the _up metric to be reliable
- add a test to show examples for your observed behavior where the _up metric has a value of falsely 0 and that it becomes stable 1.
- [optional] add a label to the _up metric to make strange situations visible. This is in fact something I still need to think about and I guess this might confuse users again. So I'm open for your thoughts. Maybe this isn't required anymore when the _up metric is more reliable.
